extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -74,6 +74,24 @@ const Direction = {
 
 const MIN_AUTO_ADVANCE_INTERVAL = 1000;
 
+const preventNavigation = (event) => {
+  // Center point of the touch area
+  const touchXPosition = event.touches[0].pageX;
+  // Size of the touch area
+  const touchXRadius = event.touches[0].radiusX || 0;
+
+  // We set a threshold (10px) on both sizes of the screen,
+  // if the touch area overlaps with the screen edges
+  // it's likely to trigger the navigation. We prevent the
+  // touchstart event in that case.
+  if (
+    touchXPosition - touchXRadius < 10 ||
+    touchXPosition + touchXRadius > window.innerWidth - 10
+  ) {
+    event.preventDefault();
```

Comment:
Two comments here:

1. Active touch actions would cause some degradation of CPU.
2. It'd be hard to make this work to correctly work with vertical and horizontal scrolling.